### ISSUE TITLE
deepMap, setKey, add support for nested arrays in 'key' argument

### ISSUE THREAD

**Issue by 54vanya** [nanostores/nanostores#253]
version 0.9.4

working example
```
import { deepMap } from 'nanostores';

type Test = {
    test: number[][];
};

export const $test = deepMap<Test>({
    test: [
        [1, 2],
        [3, 4],
    ],
});

export function setTestArrayItem() {
    console.log(JSON.stringify($test.get()));
    $test.setKey('test[1][0]', 0);
    console.log(JSON.stringify($test.get()));
}
```
result is 
```
{"test":[[1,2],[3,4]]}
{"test":[[1,2],[3,4]],"test[1]":[0]}
```
workaround
```
$test.setKey('test[1].0', 0);
$test.setKey('test.1[0]', 0);
$test.setKey('test.1.0', 0);
```

**Comment by ai** [comment]
Can I ask you to send PR?

Both I and @dkzlv is busy right now.

**Comment by ai** [comment]
Thanks. Released in 0.9.5.
